sdk/cosmosdb/azure_data_cosmos/src/options/mod.rs

```diff
@@ -0,0 +1,39 @@
+use azure_core::ClientOptions;
+
+#[derive(Clone, Debug, Default)]
+pub struct CosmosClientOptions {
+    pub(crate) client_options: ClientOptions,
+}
+
+impl CosmosClientOptions {
+    pub fn builder() -> builders::CosmosClientOptionsBuilder {
+        builders::CosmosClientOptionsBuilder::default()
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct ReadDatabaseOptions {}
+
+impl ReadDatabaseOptions {}
+
+pub mod builders {
+    use crate::{CosmosClientOptions, ReadDatabaseOptions};
+
+    #[derive(Default)]
+    pub struct CosmosClientOptionsBuilder(CosmosClientOptions);
+
+    impl CosmosClientOptionsBuilder {
```

Comment:
Currently, this PR doesn't contain any way to _set_ the `client_options`. That's because we need some sort of pattern to be established in `azure_core`. I'm looking at options for that (derive macros, traits, whatnot) but for now I'm just leaving it out with the intent that it will come later via ✨ New Cool Magic ✨ .

Reply:
#1737 will define a derive macro, but we don't want users to set `client_options` directly (the [guidelines are a little ahead on this](https://azure.github.io/azure-sdk/rust_introduction.html#rust-client-configuration)). You need to implement the `ClientOptionsBuilder` for now as generated code does: <https://github.com/Azure/autorest.rust/blob/a27439cd9817c41aa801c89534ec4c873757c851/packages/typespec-rust/test/sdk/keyvault_secrets/src/generated/clients/key_vault_client.rs#L249>.